src/css/transformHEXRGBa.ts

```diff
@@ -11,17 +11,62 @@ export function transformHEXRGBaForCSS(token: DesignToken): DesignToken['value']
   const type = token.$type ?? token.type;
   if (val === undefined) return undefined;
 
-  const transformHEXRGBa = (val: string) => {
-    const regex = /rgba\(\s*(?<hex>#.+?)\s*,\s*(?<alpha>\d*(\.\d*|%)*)\s*\)/g;
-    return val.replace(regex, (match, hex, alpha) => {
-      try {
-        const [r, g, b] = new Color(hex).srgb;
-        return `rgba(${r * 255}, ${g * 255}, ${b * 255}, ${alpha})`;
-      } catch (e) {
-        console.warn(`Tried parsing "${hex}" as a hex value, but failed.`);
-        return match;
+  const transformHexColor = (hex: string) => {
+    try {
+      // Fast path for invalid hex
+      if (hex.length < 4) return hex;
+
+      // Determine format based on length
+      const hexLength = hex.length - 1; // subtract 1 for #
+
+      // Only transform hex colors with alpha channel
+      const hasAlpha = hexLength === 4 || hexLength === 8;
+      if (!hasAlpha) return hex;
+
+      let hexColor = hex;
+      let alpha = '1';
+
+      // Convert shorthand to full format if necessary
+      if (hexLength === 4) {
+        const r = hex[1],
+          g = hex[2],
+          b = hex[3],
+          a = hex[4];
+        hexColor = `#${r}${r}${g}${g}${b}${b}`;
+        alpha = (parseInt(a + a, 16) / 255).toString();
+      } else if (hexLength === 8) {
+        alpha = (parseInt(hex.slice(7), 16) / 255).toString();
+        hexColor = hex.slice(0, 7);
       }
-    });
+
+      const [r, g, b] = new Color(hexColor).srgb;
+      return `rgba(${r * 255}, ${g * 255}, ${b * 255}, ${alpha})`;
+    } catch (e) {
+      return hex;
+    }
+  };
+
+  const transformHEXRGBa = (val: string) => {
+    // Handle standalone hex colors
+    if (val.startsWith('#')) {
+      return transformHexColor(val);
+    }
+
+    // Handle rgba() with hex colors
+    if (val.includes('rgba(')) {
+      return val.replace(/rgba\(\s*#[A-Fa-f0-9]+\s*,\s*([0-9.%]+)\s*\)/g, (match, alpha) => {
+        const hex = match.substring(match.indexOf('#'), match.indexOf(','));
+        try {
+          const [r, g, b] = new Color(hex).srgb;
+          return `rgba(${r * 255}, ${g * 255}, ${b * 255}, ${alpha})`;
```

Comment:
What about `rgba(hex8, alpha)` or `rgba(hex4, alpha)` so basically which alpha do we take if they are both defined in the hex but also in the alpha component of the expression?